.github/workflows/go-tests.yml

```diff
@@ -1,17 +1,47 @@
-name: Tests
+name: Build services
 
 on:
-  workflow_call:
+  push:
+    paths:
+      - "cmd/users/**"
+      - "cmd/stellar-dominion/**"
+      - "pkg/**"
+      - "internal/**"
+      - ".github/workflows/go-tests.yml"
+      - ".github/workflows/services*.yml"
+      - ".github/workflows/user-service*.yml"
+      - ".github/workflows/stellar-dominion-service*.yml"
```

Comment:
The main workflow reacts on:
* any change to the CI workflows.
* any change in the code.

This makes it easier to orchestrate if a single workflow is reacting instead of multiple being triggered in parallel.